docs/dev/replay-exec.md

```diff
@@ -0,0 +1,86 @@
+This doc describes how to dump an Exec runtime meta and column batch data to a directory,
+and replay the Exec with the dumped meta and data. When encountering a perf issue using this tool
+can dump the runtime meta/data and then replay. It will spend less time when collecting NSYS and
+NCU information when replaying the dumped runtime meta/data.
+
+# Dump a Exec meta and a column batch data
+
+## compile Spark-Rapids jar
+e.g.: compile for Spark330
+```
+mvn clean install -DskipTests -pl dist -am -DallowConventionalDistJar=true -Dbuildver=330 
+```
+Note: Should specify `-DallowConventionalDistJar`, this config will make sure to generate a
+conventional jar. If you do not specify this config, then replay will fail because of can not
+find the class `com.nvidia.spark.rapids.debug.ProjectExecReplayer`
+
+## enable dump
+This assumes that the RAPIDs Accelerator has already been enabled.   
+This feature is disabled by default.   
+Set the following configurations to enable this feature:
+
+``` 
+spark.conf.set("spark.rapids.sql.debug.replay.exec.types", "project")
+```
+Default `types` value is empty which means do not dump.   
+Define the Exec types for dumping, separated by comma, e.g.: `project,aggregate,sort`.   
```

Comment:
can you specify for different types of joins and other execs - like broadcastjoin vs hashjoin?